File: problems/002/002.f90
PROGRAM F002

    ! Copyright 2021 Melwyn Francis Carlo

    IMPLICIT NONE

    INTEGER :: N
    REAL    :: BINETS_FORMULA, FN_MAX, FN_SUM, FN_SUM_EVEN

    FN_MAX = 4E+6
    N      = INT( LOG((FN_MAX * SQRT(5.0)) + 0.5) / LOG((1 + SQRT(5.0)) / 2) )
    FN_SUM = (BINETS_FORMULA(N + 2) - 1)

    FN_SUM_EVEN = FN_SUM / 2;

    PRINT ('(I0)'), INT(FN_SUM_EVEN)

END PROGRAM F002


REAL FUNCTION BINETS_FORMULA (N_VAL)

    IMPLICIT NONE

    INTEGER, INTENT(IN) :: N_VAL

    BINETS_FORMULA = REAL( (1 / SQRT(5.D0))                                    &
                     * ( (((1 + SQRT(5.D0)) / 2) ** N_VAL)                     &
                        - (((1 - SQRT(5.D0)) / 2) ** N_VAL) ), 4)

END FUNCTION BINETS_FORMULA

File: problems/002/002.c
#include <stdio.h>
#include <math.h>

long int binets_formula(int n_val) {
    return (1 / sqrt(5)) * (pow((1+sqrt(5)) / 2, n_val) 
    - pow((1-sqrt(5)) / 2, n_val));
}

int main() {
    int Fn_max = 4E+6;
    int n = ( log((Fn_max * sqrt(5)) + 0.5) / log((1 + sqrt(5)) / 2) );
    long int Fn_sum = binets_formula(n+2) - 1;
    long int Fn_sum_even = Fn_sum / 2;
    printf("\n%ld\n\n", Fn_sum_even);
    return 0;
}
